### ISSUE TITLE
deSawyer: removing the dependency on Sawyer library and clean up utility 

### ISSUE THREAD

**Issue by yanyh15** [passlab/rexompiler#152]
Sawyer provides several utilities for ROSE, e.g. command line processing, diagnostics/messaging/logging, string utility, file system, assert, graph support, etc. However, in the whole source base, there are also files and implementations that provide full or partial functions of those utilities. For this effort, we will remove the usage of Sawyer (along with those in src/Rose folder), cleanup usage of them in other places of the source tree, simplify the sources and usage of functions for those required utilities. More specifically: 
1. diagnostics/messaging/logging: replaced with APIs such as MLOG_FATAL_C, MLOG_FATAL_CXX, ..., MLOG_INFO_C, MLOG_FATAL_CXX, which are macros that allow for easy inclusion of __FILE__:__LINE__:__func__ info in the mlogging msgs.
2. command line processing: revert back to the original ROSE usage of command line processing (in src/frontend/SageIII/sage_support. We will evaluate the implementation later on to see it needs to be improved. 
3. String utilities: 
4. File system:
5. Assert: 
6. Graph support:

**Comment by yanyh15** [comment]
addressed by this https://github.com/passlab/rexompiler/pull/154 and now the building time can come down below 20m on github.

**Comment by ouankou** [comment]
The building is super fast now! However, I happen to notice a problem. According to the CI testing log, there are a lot of warnings. These messages start with "ERROR ..." or "WARN ...". I just ran tests on checked the code before this pull request. These warnings didn't show up.

This log can be found here: https://github.com/passlab/rexompiler/actions/runs/4017044154/jobs/6900877656
I also downloaded the raw log and attached it here: [ci_test.log](https://github.com/passlab/rexompiler/files/10514812/ci_test.log)

Some examples I collected from my local run are as follows. They are very similar to the ones in the CI test.

```bash
ERROR: sage_support	[../../../../rex_src/src/frontend/SageIII/sage_support/sage_support.cpp:6519,	getAssociatedFunctionSymbol]: There should be no other cases functionExp = 0x7fd262eaa470 = SgNonrealRefExp 
Inside of Sg_File_Info::display(In SgFunctionCallExp::getAssociatedFunctionSymbol(): case not supported: debug) of this pointer = 0x7fd25e35cd50 
     isTransformation                      = false 
     isCompilerGenerated                   = false 
     isOutputInCodeGeneration              = false 
     isShared                              = true (shared within merged AST) 
     isFrontendSpecific                    = false 
     isSourcePositionUnavailableInFrontend = false 
     isCommentOrDirective                  = false 
     isToken                               = false 
     isDefaultArgument                     = false 
     isImplicitCast                        = false 
     (computed) filename = /usr/include/c++/9/bits/vector.tcc 
     (computed) line     = 313  column = 6 
     (internal) file_id   = 106 
     (internal) filename  = /usr/include/c++/9/bits/vector.tcc 
     (internal  line      = 313  column   = 6 
     (computed) physical_file_id    = 106 = /usr/include/c++/9/bits/vector.tcc 
     (internal) physical_file_id    = 106 = /usr/include/c++/9/bits/vector.tcc 
     (internal) physical_file_id    = 106 = /usr/include/c++/9/bits/vector.tcc 
     (computed) physical_line       = 313 
     source_sequence_number         = 44461
...
WARN : astDiagnostics	[../../../rex_src/src/midend/astDiagnostics/AstConsistencyTests.C:2696,	visit]: definingDeclaration_access_modifier         = 3 
WARN : astDiagnostics	[../../../rex_src/src/midend/astDiagnostics/AstConsistencyTests.C:2697,	visit]: firstNondefiningDeclaration_access_modifier = 4 
WARN : astDiagnostics	[../../../rex_src/src/midend/astDiagnostics/AstConsistencyTests.C:2695,	visit]: (different access modifiers used) definingDeclaration = 0x7fd2631a6ff0 firstNondefiningDeclaration = 0x7fd2631a6bc0 = SgTemplateInstantiationDecl
...
WARN : astDiagnostics	[../../../rex_src/src/midend/astDiagnostics/AstConsistencyTests.C:5376,	visit]: TestChildPointersInMemoryPool::visit(). SgVariableSymbol is not in parent's child list, node: 0x55e6cbf5faa0 = SgVariableSymbol = user_modi_symbol_ parent: 0x55e6cbde4b60 = SgBasicBlock 
WARN : astDiagnostics	[../../../rex_src/src/midend/astDiagnostics/AstConsistencyTests.C:5253,	visit]: TestChildPointersInMemoryPool::visit(): Node is not in parent's child list, node: 0x7fc376ffe628 = SgInitializedName = initialized_name_user_modi parent: 0x55e6cbde4b60 = SgBasicBlock
```

**Comment by yanyh15** [comment]
@ouankou Yes, I am aware of that. It is due to the way verbose level is handled by ROSE, which is still a mess so far. I will raise a ticket for it. When I changed the Sawyer-based mlog/mprintf (messaging and logging) calls in the source base, those marked with warning and error in Sawyer-based mlog/mprintf are all changed to the new API and implementation MLOG_WARN_* and MLOG_ERROR_* I added (in src/util/mlog.h/C file). Sawyer-based mlog/mprintf calls do not spit out messages for warning and errors. But for my implementation of MLOG_WARN/ERROR, the messages are all spit out. That is why you do not see it before the change and now you see it after we change. This need a cleanup to use of MLOG_* facility correctly. Right now, I have not change that. It is not urgent. If we want to turn of those messages, find those file:line number according to the messages, and change MLOG_WARN/ERROR_* to MLOG_KEY_* should shut up those messages.
